mlir/lib/Dialect/Vector/Transforms/VectorTransforms.cpp

```diff
@@ -1771,6 +1771,14 @@ struct DropUnitDimsFromTransposeOp final
       newPerm.push_back(idx - droppedDimsBefore[idx]);
     }
 
+    // Fixup for `newPerm`. The `sourceTypeWithoutUnitDims` could be vector<1xT>
+    // type when the dimensions are unit dimensions. In this case, the newPerm
+    // should be [0].
+    if (sourceTypeWithoutUnitDims.getRank() == 1 &&
+        sourceTypeWithoutUnitDims.getShape()[0] == 1 && newPerm.empty()) {
```

Comment:
Could this just be `newPerm.empty()`. I think that's only possible in the all-unit dim case.

Reply:
Yes, it could be. My concern was that it could be wrong if someone updates `dropNonScalableUnitDimFromType`. The test should be able to capture the case I guess, so I'm okay with just checking `newPerm.empty()`.

Reply:
I think the test is okay :+1: 

Btw, to check for a unit dim you would have to do `sourceDims[0] == std::make_tuple(1, false)` (to check the scalability too).

Reply:
ah, good point!